code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/process/ServiceActivate.java

```diff
@@ -211,10 +213,16 @@ public Object call() throws Exception {
         return ntwkIds;
     }
 
-    protected Long getImage(String imageUuid) {
+    protected Long getImage(String imageUuid, Object registryCredentialId) {
```

Comment:
Hmm.  What if we casted it in createServiceInstances() and changed getImage to accept a Long?